src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1154,31 +1134,28 @@ public function getChanges()
      * Determine if the new and old values for a given key are equivalent.
      *
      * @param  string $key
-     * @param  mixed  $current
      * @return bool
      */
-    public function originalIsEquivalent($key, $current)
+    public function originalIsEquivalent($key)
     {
         if (! array_key_exists($key, $this->original)) {
             return false;
         }
 
+        $attribute = $this->getAttribute($key);
         $original = $this->getOriginal($key);
 
-        if ($current === $original) {
+        if ($attribute === $original) {
```

Comment:
I think is better check fail case first. 
```
if (is_null($attribute)) {
    return false;
} elseif ($attribute === $original) { 
    return true;
} elseif ($this->isDateAttribute($key)) {
    return $this->fromDateTime($attribute) === $this->fromDateTime($original);
}
```
originalIsEquivalent must return `true` if `$original` is null and `$attribute` is null too ?

Reply:
I agree, but it is already existing code and beyond the scope of this PR. I'm happy to refactor this method as well, but I think it should be in a different PR.

Reply:
I agree.